### ISSUE TITLE
DAR401 false positive

### ISSUE THREAD

**Issue by Trim21** [terrencepreilly/darglint#68]
```python


def false_positive() -> None:
    """summary

    :raises ValueError: description
    """
    try:
        raise ValueError("233")
    except ValueError as e:
        raise e from None

```
```
DAR401 Missing exception(s) in Raises section: -r e
```

**Comment by terrencepreilly** [comment]
Okay, I've updated error analysis to handle the catch-and-throw pattern, and to be a little more robust in general.

Unfortunately, I probably won't get it to working in 100% of cases any time soon.  The problem is that in order to capture weird patterns, I'd have to run through all assignments, etc.  That's a bit like writing an interpreter.

For example, the following will probably fail even now:

```
def f():
    """Something.

    Raises:
        Exception: Always.

    """
    e = Exception()
    raise e
```

Although it's relatively simple, you can imagine analysis getting complicated (especially in classes, where you could have stuff like

```
class A:
    def set_error(self):
        self.error = IOError()

    def raise_error(self):
        """Raises an error.

        Raises:
            IOError:  Always, but darglint will report an error here.

        """
        raise self.error
```

which I'll probably never end up handling -- in fact, it may not be *possible* to handle this except at runtime.)

**Comment by matthieucan** [comment]
Hello,
It's really nice that this feature allows to catch this pattern:
```python
try:
    foo
except Exception as e:
    do something
    raise e
```

However it doesn't catch this one:
```python
try:
    foo
except Exception:
    do something
    raise
```

Would it require many changes to also have this? I think this pattern is common enough (mentioned in the Python docs https://docs.python.org/3/tutorial/errors.html) that many people would benefit from it.
Thank you in advance!

**Comment by terrencepreilly** [comment]
Good point, that should be pretty easy.

**Comment by matthieucan** [comment]
Thank you for answering so fast and reopening! Happy to test the changes later :)

**Comment by terrencepreilly** [comment]
This fix should be present now in v1.2.3

**Comment by KyleKing** [comment]
I upgraded to `1.2.3` and the test example works for me where there is an explicit raise in the `try` block:

https://github.com/terrencepreilly/darglint/blob/137d6bb1508a143b19a1a136bb410324b1f0cc82/tests/test_integrity_checker.py#L797-L810

However, when the error that is caught is coming from a different function or library, there are DAR401 and DAR402 errors Is this something that could be handled?

```py
import json
from json.decoder import JSONDecodeError


def false_positive():
    """Test bare raise.

    Raises:
        JSONDecodeError: description

    """
    try:
        json.loads('[...]')
    except JSONDecodeError:
        raise
```

**Comment by terrencepreilly** [comment]
@KyleKing, I didn't see your comment here.  See Issue #88 for its resolution, though.  It should be fixed when 1.3.1 is released.
